sycl/include/std/experimental/simd.hpp

```diff
@@ -1684,7 +1684,7 @@ class simd_mask {
 #ifdef ENABLE_SYCL_EXT_ONEAPI_INVOKE_SIMD
   // TODO inefficient, use this's and __v's storage directly
   template <class _Up>
-  simd_mask(const simd_mask<_Up, simd_abi::fixed_size<size()>>& __v) noexcept {
+  simd_mask(const simd_mask<_Up, abi_type>& __v) noexcept {
```

Comment:
I'm not sure about this change.

TS2 shows that this constructor is only available from fixed-sized masks: https://en.cppreference.com/w/cpp/experimental/simd/simd_mask/simd_mask

Reply:
Thanks for the review. The problem I'm trying to fix is the implicit conversion constructor doesn't work. Mind helping me find a better fix?

test.cpp:
```
#include <std/experimental/simd.hpp>

namespace simd_abi {
template <class T, int N>
using native_fixed_size = typename std::experimental::__simd_abi<
    std::experimental::_StorageKind::_VecExt, N>;
}

template <class T, int N>
using simd_mask =
    std::experimental::simd_mask<T, simd_abi::native_fixed_size<T, N>>;

int main() {

simd_mask<int, 8> x;
simd_mask<double, 8> y(x);
}

```

error:
```
test.cpp:16:22: error: no matching constructor for initialization of 'simd_mask<double, 8>' (aka 'simd_mask<double, __simd_abi<std::experimental::_StorageKind::_VecExt, 8>>')
simd_mask<double, 8> y(x);
                     ^ ~
../include/std/experimental/simd.hpp:1648:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'simd_mask<int, [...]>' to 'const simd_mask<double, [...]>' for 1st argument
class simd_mask {
      ^
../include/std/experimental/simd.hpp:1648:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'simd_mask<int, [...]>' to 'simd_mask<double, [...]>' for 1st argument
../include/std/experimental/simd.hpp:1674:12: note: candidate constructor not viable: no known conversion from 'simd_mask<int, 8>' (aka 'simd_mask<int, __simd_abi<std::experimental::_StorageKind::_VecExt, 8>>') to 'value_type' (aka 'bool') for 1st argument
  explicit simd_mask(value_type __v) noexcept {
           ^
../include/std/experimental/simd.hpp:1687:3: note: candidate template ignored: could not match '__simd_abi<1, [...]>' against '__simd_abi<2, [...]>'
  simd_mask(const simd_mask<_Up, simd_abi::fixed_size<size()>>& __v) noexcept {
  ^
../include/std/experimental/simd.hpp:1669:3: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
  simd_mask() = default;
  ^
../include/std/experimental/simd.hpp:1699:3: note: candidate constructor template not viable: requires 2 arguments, but 1 was provided
  simd_mask(const value_type*, _Flags);
  ^
1 error generated.
```

Reply:
Ah, I see...  We should wait to hear from @rolandschulz (because he's more familiar with the `std::simd` proposal), but I would be tempted to add a _new_ constructor alongside the existing one, rather than replacing it.  So, we'd have both:

```c++
simd_mask(const simd_mask<_Up, simd_abi::fixed_size<size()>>& __v) noexcept;
simd_mask(const simd_mask<_Up, abi_type>& __v) noexcept;
```

Does that work?

Reply:
ah yeah thats a better idea thanks, let me tentatively implement that